packages/synapse-react-client/src/components/Plot/SynapsePlot.tsx

```diff
@@ -1,135 +1,165 @@
-import Plotly from 'plotly.js-basic-dist'
+import Plotly, { AxisType, PlotType } from 'plotly.js-basic-dist'
 import React from 'react'
 import createPlotlyComponent from 'react-plotly.js/factory'
 import { SynapseConstants } from '../../utils'
-import { getFullQueryTableResults } from '../../synapse-client/SynapseClient'
 import {
-  QueryResultBundle,
+  FacetColumnRequest,
   QueryBundleRequest,
+  QueryFilter,
+  Row,
 } from '@sage-bionetworks/synapse-types'
 import { parseEntityIdFromSqlStatement } from '../../utils/functions/SqlFunctions'
-import { SynapseContext } from '../../utils/context/SynapseContext'
+import { useGetFullTableQueryResults } from '../../synapse-queries'
+import { Skeleton } from '@mui/material'
+import { QueryWrapperSynapsePlotRowClickEvent } from '../QueryWrapperPlotNav/QueryWrapperSynapsePlot'
+import { QueryContextType } from '../QueryContext'
 const Plot = createPlotlyComponent(Plotly)
 
-export type SynapsePlotProps = {
-  ownerId?: string
-  wikiId?: string
-  widgetparamsMapped?: any
+export type SynapsePlotWidgetParams = {
+  query: string //sql string
+  title?: string //plot title
+  xtitle?: string // x-axis title
+  ytitle?: string // y-axis title
+  type: string // Plotly PlotType
+  xaxistype?: string // Plotly AxisType
+  showlegend?: string // sets the legend visibility ('true' | 'false')
+  horizontal?: string // sets the if a bar chart should be horizontal or vertical ('true' | 'false')
+  barmode?: string // Plotly barmode ('stack' | 'group' | 'overlay' | 'relative')
+  displayModebar?: string // sets the modebar visibility ('true' | 'false')
 }
 
-type SynapsePlotState = {
-  isLoaded: boolean
-  queryData: QueryResultBundle
+// QueryWrapperPlotNav customPlot parameters, undefined otherwise
+export type QueryWrapperPlotNavCustomPlotParams = {
+  selectedFacets: FacetColumnRequest[]
+  additionalFilters: QueryFilter[]
+  onCustomPlotClick?: (event: QueryWrapperSynapsePlotRowClickEvent) => void
+  queryContext?: QueryContextType
+}
+export type SynapsePlotProps = {
+  widgetparamsMapped: SynapsePlotWidgetParams
```

Comment:
Consider creating a new component that wraps SynapsePlot that has the `widgetparamsMapped` interface (`SynapseWikiPlot`?) and is only used by the wiki. Then you could directly use the strict/accurate Plotly types in `SynapsePlotProps` and not have to use this fairly type-unsafe `widgetparamsMapped` intermediate data structure in places like the new QueryWrapperSynapsePlot

Reply:
Great idea.  I think the existing "MarkdownSynapsePlot" wrapper component could service this purpose.  Moved string param resolution to that component.